hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -207,7 +207,7 @@ private void initMetadataReader() {
   }
 
   private HoodieMetadataFileSystemView getMetadataView() {
-    if (metadataView == null) {
+    if (metadataView == null || !metadataView.equals(metadata.getMetadataFileSystemView())) {
```

Comment:
I am not sure I get this fix. our Metadata writers are shotlived in general. We just create a new writer, initial new partitions, apply updates for a given commit and close it out. 
So, how come metadataView can be out of sync ?

Reply:
Without much context: lets not do any fancy optimizations yet or handle in vs out of sync

Reply:
sure. we understand the issue. myself and sagar are aligned on the fix here.